### PR TITLE
Cleanup tasks in SecurityManager/SetupEncryptionStore

### PR DESCRIPTION
**Pull request by richvdh**
A few small cleanup tasks. Review commit-by-commit.

### CODE REVIEW

File: src/SecurityManager.ts [hunk 1]
```diff
@@ -313,10 +309,8 @@ export async function withSecretStorageKeyCache<T>(func: () => Promise<T>): Prom
     } finally {
         // Clear secret storage key cache now that work is complete
         secretStorageBeingAccessed = false;
-        if (!isCachingAllowed()) {
-            secretStorageKeys = {};
-            secretStorageKeyInfo = {};
-        }
```
Comment (t3chguy) [inline src/SecurityManager.ts:319]: why does this no longer need to check `secretStorageBeingAccessed`

**Review by andybalaam — APPROVED** [review]
Same question as t3chguy, all good otherwise

File: src/SecurityManager.ts (hunk 1 above)
Comment (richvdh) [reply src/SecurityManager.ts:319]: because the line just before this is:

```
        secretStorageBeingAccessed = false;
```

Comment (t3chguy) [reply src/SecurityManager.ts:319]: Ah yes, so it was redundant all along

**Review by t3chguy — APPROVED** [review]

File: src/SecurityManager.ts (hunk 1 above)
Comment (richvdh) [reply src/SecurityManager.ts:319]: since https://github.com/matrix-org/matrix-react-sdk/pull/4789, yes

**Pull request merged**